### DISCUSSION TITLE
archives.rlcp and source.rlcp

### DISCUSSION THREAD

**Discussion by caarlos0** [goreleaser/goreleaser#3659] [Ideas, Issue Triage, and General discussions]
This is a place to discuss the relative longest common path changes made recently in `source` and `archives`:

- https://goreleaser.com/deprecations/#archivesrlcp
- https://goreleaser.com/deprecations/#sourcerlcp

Related PR and issue:
- https://github.com/goreleaser/goreleaser/pull/3656
- https://github.com/goreleaser/goreleaser/issues/3655

If you think it shouldn't be default, or have an use case that cannot be worked through with it, this is the place to let us know!

**Comment by at-wat** [comment]
I don't have problem on the change but had confusion about the warning message.

I have config like
```yaml
archives:
  - files:
      - script.sh
```
and got 
```
  • setting defaults
      • DEPRECATED: `archives.rlcp` will be the default soon, check https://goreleaser.com/deprecations#archivesrlcp for more info
...
  • your config is using deprecated properties, check logs above for details
```
(https://github.com/at-wat/goreleaser-repro-rlcp [check result](https://github.com/at-wat/goreleaser-repro-rlcp/actions/runs/3834378994/jobs/6526724005#step:3:16))

I checked generated tarballs with `rlcp: true/false` and without `rlcp` option and this config seems not affected.

**Reply by caarlos0** [reply]
It's like this old xkcd: 

![](https://imgs.xkcd.com/comics/workflow.png)

The current behavior (`rlcp: false`) is weird, and probably wrong for some cases, but no doubt there will be someone depending on it.

In most cases (including yours), the change seems to have no effect, so the recommendation is to just set it to true and forget about it :)

Hopefully that clarifies things a little bit.

**Comment by caarlos0** [comment]
FWIW on v1.19 this will be default! 

So, if you set rlcp in your settings, you can now remove it.

If you don't, now its a good time to test it. I want to release v1.19 really soon!
